src/Units/UnitContext.java

```diff
@@ -196,6 +196,6 @@ public void chooseWeapon(ITargetable targetType, int range, boolean afterMoving)
   @Override
   public String toString()
   {
-    return model.name;
+    return "Context for "+model.name+" at "+coord+", using weapon "+weapon;
```

Comment:
Better, but can't weapon (and coord?) be null? Or the context may have been created just to calculate a price difference, which isn't reflected here at all?

Reply:
They can be, but `null`'s string concatenation is the empty string rather than a crash.

That's fair, but I don't really feel like adding a read-back of literally every field.
If you really want it, then I suggest we start using Lombok https://projectlombok.org/features/all

Reply:
It's not that I really want a more complex toString; it's more that I want a less complex `UnitContext`.

If wishes were fishes, I know.